addons/mail/static/src/js/client_action.js

```diff
@@ -287,6 +356,41 @@ var ChatAction = Widget.extend(ControlPanelMixin, {
                 self.thread.scroll_to({offset: offset});
             });
     },
+    /**
+     * @private
+     */
+    _onModeration: function (messageID, decision) {
+        if (decision === 'reject') {
+            this._onRejectMessage([messageID]);
+        } else if (decision === 'discard') {
+            this._onDiscardMessage([messageID]);
+        } else if (decision === 'ban') {
+            this._onBan(messageID);
+        } else {
+            chat_manager.moderate([messageID], decision);
+        }    
+    },
+
+    _onDiscardMessage: function (messageIDs) {
```

Comment:
missing docstring, this is a private method, not a handler (should be named _discardMessage)